File: designPattern/factory/python/Factory.py
import abc

class Factory:
	__metaclass__ = abc.ABCMeta

	def create(self, company, quality):
		p = self.createProduct(company, quality)
		self.registProduct(p)
		return p

	@abc.abstractmethod
	def createProduct(self, company, quality):
		pass

	@abc.abstractmethod
	def registProduct(self, product):	
		pass

File: designPattern/factory/review/java/frameWork/Factory.java
package frameWork;

public abstract class Factory{

	public final Product create(String company, String phoneName, String ficture){
		Product p = createProduct(company, phoneName, ficture);
		registerProduct(p);
		return p;
	} 

	protected abstract Product createProduct(String company, String phoneName, String ficture);
	protected abstract void registerProduct(Product product);
}